src/project/directory.rs

```diff
@@ -0,0 +1,104 @@
+
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::io::Read;
+use project::ProjectError;
+
+/// Tracking root folder for files as well as a list of files in destination.
+#[derive(Debug)]
+pub struct Directory {
+    /// Root directory of the file system
+    pub root: PathBuf,
+
+    /// Whether recursively check subfolders
+    ///
+    /// non-recursive:
+    /// ```
+    /// resources/
+    ///   config/
+    ///     config.yml
+    ///     display.yml
+    ///     logging.yml
+    /// ```
+    ///
+    /// recursive:
+    /// ```
+    /// resources/
+    ///   config/
+    ///     config.yml
+    ///     display/
+    ///       config.yml # same as "resources/config/display.yml"
+    ///     logging.yml
+    /// ```
+    pub recursive: bool,
+
+    /// Extensions of files it will look for.
+    pub extensions: Vec<String>,
+
+    /// list of files found
+    pub list: Vec<PathBuf>,
+}
+
+impl Directory {
+    /// Create a new directory from a path and extensions, recursive will search for files
+    /// with the extension recursively.
+    pub fn new<P: AsRef<Path>>(path: P,
+                               recursive: bool,
+                               extensions: Vec<String>)
+                               -> Result<Directory, ProjectError> {
+        let mut directory = Directory {
+            root: path.as_ref().into(),
+            recursive: recursive,
+            extensions: extensions,
+            list: Vec::new(),
+        };
+
+        try!(directory.refresh());
+        Ok(directory)
+    }
+
+    /// Re-check directory for yaml files and refresh contents
+    pub fn refresh(&mut self) -> Result<(), ProjectError> {
+        let list =
+            try!(Directory::files_in_dir(self.root.clone(), self.recursive, &self.extensions));
+        for entry in list {
+            self.list.push(entry.clone());
+        }
+
+        Ok(())
+    }
+
+    /// Return contents of a file.
+    pub fn load<P: AsRef<Path>>(path: P) -> Result<String, ProjectError> {
+        let mut file = try!(fs::File::open(&path));
```

Comment:
@Aceeri I first thought about android assets, but since this is will also be for mutated - if I understand correctly - Android assets are not suitable, so the current solution works just fine (because we have to use file access on Android then, anyways) :tada: